.gitmodules

```diff
@@ -1,3 +1,3 @@
 [submodule "src/externals/paho-mqtt-c"]
 	path = src/externals/paho-mqtt-c
-	url = https://github.com/eclipse/paho.mqtt.c.git
+	url = ../paho.mqtt.c.git
```

Comment:
Does this break forks for people who have only the CPP version forked? 
The solution to it is just to fork the c lib yourself. It was trying to pull down my own fork which I had not done.

Reply:
@Zelif is correct. The _main point_ of using the submodule is so that people can just grab the Paho C++ code and build it without having to think or worry about the C lib. 

That would mean grabbing the C lib directly off of GitHub... not from a local directory! It would be wrong to assume that the C library sources have been downloaded locally.

Reply:
No. `../..` refers to `github.com`, and it will use the same protocol as the parent module. This is to avoid using `https` submodule even if you clone the parent module with `ssh`.

Reply:
The reason `..` doesn't work when you fork it is that this repository will then be in your own namespace. If someone fork the current repository, the full repository path of the forked repository will be `github.com/{username}/paho.mqtt.cpp`, and `..` will refer to `github.com/{username}`.

Reply:
https://git-scm.com/docs/git-submodule#_commands
You can have a look at doc about relative submodule URL.